File: native/mixdog-graph/tests/fixtures/calls/java/sample.java
// hidden()

class Chain {
  Chain b() { return this; }
  Chain c() { return this; }
}

class Widget {
  static String quoted = "hidden()";
  static int g = plain();

  static int plain() { return 0; }
  static void inner() {}
  static int leaf(int x) { return x; }
  static void nest(int x) {}
  static void helper() {}
  static void seed() {}

  void ping() {}

  static void run(Chain a) {
    inner();
    nest(leaf(1));
    a.b().c();
    new Widget();
  }

  void act() {
    helper();
    this.ping();
  }

  static void seedLine() {
    String mark = "μ"; seed();
  }
}

File: native/mixdog-graph/tests/fixtures/calls/rust/sample.rs
#![allow(dead_code, unused_variables, unused_must_use)]

// hidden()
const QUOTED: &str = "hidden()";

const fn plain() -> i32 {
    0
}

const T: i32 = plain();

fn inner() {}
fn leaf(x: i32) -> i32 { x }
fn nest(x: i32) { let _ = x; }
fn helper() {}
fn seed() {}

struct Chain;
impl Chain {
    fn b(self) -> Self { self }
    fn c(self) -> Self { self }
}

fn run(a: Chain) {
    inner();
    nest(leaf(1));
    a.b().c();
}

struct Widget;
impl Widget {
    fn new() -> Self { Widget }
    fn ping(&self) {}
    fn act(&self) {
        helper();
        self.ping();
    }
}

fn main() {
    let mark = "μ"; seed();
    println!("{}", T);
    let _ = Widget::new();
}
